server/src/test/java/io/crate/integrationtests/KnnIntegrationTest.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.integrationtests;
+
+import static io.crate.testing.Asserts.assertThat;
+
+import java.util.ArrayList;
+
+import org.elasticsearch.test.IntegTestCase;
+
+import io.crate.testing.UseJdbc;
+
+
+public class KnnIntegrationTest extends IntegTestCase {
+
+    // does only work with http, but not with postgres protocol
+    @UseJdbc(0)
+    public void test_insert_select_and_match_2048_float_vector() {
+        int size = 2048;
+        execute("create table tbl(x float_vector(" + size + "));");
+        var values = new ArrayList<String>();
+        for (int i = 0; i < size; i++) {
+            values.add(Float.toString(randomFloat()));
+        }
+        var valuesAsString = "[" + String.join(",", values) + "]";
+        // does not work as an argument for a prepared statement
+        execute("insert into tbl(x) values (" + valuesAsString + ")");
```

Comment:
```suggestion
        execute("create table tbl(x float_vector(2048));");
        float[] vector = new float[2048];
        for (int i = 0; i < vector.length; i++) {
            vector[i] = randomFloat();
        }
        execute("insert into tbl(x) values (?)", new Object[] { vector });
```

The string conversions shouldn't be necessary and using float[] has a lower memory footprint.